processing/src/main/java/io/druid/query/filter/InDimFilter.java

```diff
@@ -0,0 +1,45 @@
+package io.druid.query.filter;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.metamx.common.StringUtils;
+
+import java.nio.ByteBuffer;
+import java.util.List;
+
+public class InDimFilter implements DimFilter
+{
+  private final List<String> values;
+  private final String dimension;
+
+  @JsonCreator
+  public InDimFilter(@JsonProperty("dimension") String dimension, @JsonProperty("values") List<String> values)
+  {
+    this.values = values;
+    this.dimension = dimension;
+  }
+
+  public String getDimension()
+  {
+    return dimension;
+  }
+
+  public List<String> getValues()
+  {
+    return values;
+  }
+
+  @Override
+  public byte[] getCacheKey()
+  {
+
+    byte[] dimensionBytes = StringUtils.toUtf8(dimension);
+    byte[] valueBytes = (values == null) ? new byte[]{} : StringUtils.toUtf8(values.toString());
```

Comment:
We want to guarantee that the cache key bytes and the thing we want to be the cache key have a one-to-one correspondence, to prevent false hits and false misses. Like @drcrallen said, the toString of a List does not satisfy that.

For example `Lists.newArrayList("a", "b", "c").toString()` and `Lists.newArrayList("a", "b, c").toString()` are both `[a, b, c]`.